packages/expo-modules-core/android/src/androidTest/java/expo/modules/kotlin/activityresult/ActivityResultsManagerTest.kt

```diff
@@ -0,0 +1,147 @@
+package expo.modules.kotlin.activityresult
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import androidx.activity.result.contract.ActivityResultContracts.StartActivityForResult
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.os.bundleOf
+import androidx.lifecycle.Lifecycle
+import androidx.test.core.app.ActivityScenario
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.LargeTest
+import com.google.common.truth.Truth.assertThat
+import expo.modules.kotlin.providers.CurrentActivityProvider
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+
+private val currentActivityProvider = object : CurrentActivityProvider {
+  var activity: AppCompatActivity? = null
+
+  override val currentActivity: AppCompatActivity?
+    get() = activity
+}
+private val activityResultsManager = ActivityResultsManager(currentActivityProvider)
+
+@LargeTest
+@RunWith(AndroidJUnit4::class)
```

Comment:
```suggestion
```
We should use the CPP aware test runner: https://github.com/expo/expo/blob/main/packages/expo-modules-core/android/src/androidTest/java/expo/modules/TestRunner.kt. Also, this runner is a default one. So you can commit the annotation here ;)